### ISSUE TITLE
[java] FinalFieldCouldBeStatic fails on some fields annotated with Mockito Spy

### ISSUE THREAD

**Issue by lapostoj** [pmd/pmd#5610]
**Affects PMD Version:** 

**Rule:**

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_java_design.html#finalfieldcouldbestatic>

**Description:**

**Code Sample demonstrating the issue:**

```java
    @Spy
    private final Clock clock = CLOCK;
```

where CLOCK is a static field defined somewhere else.
To note the following is not triggering the rule
```java
    @Spy
    private final Clock clock = Clock.fixed(Instant.now(), ZoneId.of("UTC"));
```

This could be related to the re-write of the rule in Java in this PR https://github.com/pmd/pmd/pull/5553

**Expected outcome:**

PMD reports a violation at line 2, but that's wrong. That's a false positive.

**Running PMD through:** *Maven*

**Comment by zbynek** [comment]
If you call the `spy` method instead of using the `@Spy` annotation, PMD  behaves correctly. 

```java
// import static org.mockito.Mockito.spy;
private final Clock clock = spy(CLOCK);
```

For correctness PMD would have to know that `@Spy` is equivalent to adding such method call to the code. I'm not involved in PMD development so I have no idea if PMD wants to go that way. Note that you can also tell PMD to skip given rule for fields annotated with `@Spy` using a config property like this:

```xml
<property name="violationSuppressXPath" value="./ancestor-or-self::FieldDeclaration[count(*/Annotation[contains(@SimpleName, 'Spy')])&gt;0]"/>
```
